### PR TITLE
[J] Workshop Jan 17

### PR DESCRIPTION
**Pull request by yannispanousis**
How far did you get?

- [x] Filled in all the `Dockerfile` placeholders
- [x] Managed to build the `Dockerfile`
- [x] Managed to run a container from it
- [x] The container stayed up and running, and Nginx showed me its startup logs (check with `docker logs nginx`)
- [x] I could access the Nginx port from the outside world
- [x] I got the page response from Nginx via the browser at http://localhost:8000 (or another port of your choosing)
- [x] I managed to push the image to Docker Hub and it looks like https://hub.docker.com/r/lystable/fuzz/

### CODE REVIEW

File: exercises/chapter2/nginx/Dockerfile [hunk 1]
```diff
@@ -1,10 +1,15 @@
-FROM <the owner of the base image>/<the name of the base image>
-MAINTAINER <your name and username>
+FROM ubuntu:latest
+MAINTAINER Jeremiah Boyle "jeremiah.boyle@gmail.com"
 
-RUN <install nginx>
-...
+ENV DEBIAN_FRONTEND noninteractive
```
Comment (yannispanousis) [inline exercises/chapter2/nginx/Dockerfile:4]: What made you add the noninteractive setting? It's probably not needed in this scenario, but definitely something used widely in Debian `Dockerfile`s!

File: exercises/chapter2/nginx/Dockerfile [hunk 2]
```diff
@@ -1,10 +1,15 @@
...
+
+RUN apt-get update && apt-get install -yq nginx
```
Comment (yannispanousis) [inline exercises/chapter2/nginx/Dockerfile:6]: Install quietly huh, that could prove quite readable!! Will try using that in some of our install scripts!

File: exercises/chapter2/nginx/Dockerfile [hunk 3]
```diff
@@ -1,10 +1,15 @@
...
 
 COPY index.html /usr/share/nginx/html/index.html
-COPY nginx.conf /etc/nginx/conf.d/nginx.conf
+COPY nginx.conf /etc/nginx/nginx.conf
+
+RUN echo "daemon off;" >> /etc/nginx/nginx.conf
```
Comment (yannispanousis) [inline exercises/chapter2/nginx/Dockerfile:11]: Now this is where things become interesting. You may have noticed that `nginx.conf` lives in the exercise directory. And in particular you'll notice we have a statement exactly above this line that says `COPY nginx.conf /etc/nginx/nginx.conf`.

What the `COPY` means, which I know we didn't dive into, is that at build-time of the `Dockerfile`, we add this file into the Docker image. Now if you're interesting in changing the content on that file for the purpose of this build, you're able to do that very easily, simply by editing the file in your host machine! Once you've changed the file, you then run the Docker build again, to update the Docker image!

Let me know this doesn't make sense.

File: exercises/chapter2/nginx/index.html [hunk 4]
```diff
@@ -1 +1 @@
-Replace me with something of your interest!
+Hello world!
```
Comment (yannispanousis) [inline exercises/chapter2/nginx/index.html:1]: Hello back!

File: exercises/chapter2/nginx/nginx.conf [hunk 5]
```diff
@@ -6,24 +6,24 @@ events {
...
 
 http {
-	##
-	# Basic Settings
-	##
+    ##
+    # Basic Settings
+    ##
 
-        access_log /dev/stdout;
```
Comment (yannispanousis) [inline exercises/chapter2/nginx/nginx.conf:13]: Tidy, good good :)

**Review by yannispanousis — COMMENTED** [review]
Looks good man! Some interesting points you came across there. Would you mind filling in the check boxes in the PR description to tell us how far you got in the exercise? Thanks!

File: exercises/chapter2/nginx/Dockerfile (hunk 3 above)
Comment (j3r3miah) [reply exercises/chapter2/nginx/Dockerfile:11]: 👍

File: exercises/chapter2/nginx/Dockerfile (hunk 1 above)
Comment (j3r3miah) [reply exercises/chapter2/nginx/Dockerfile:4]: I wasn't clear, at first, on whether `apt-get` respected noninteractive or not. Seemed like a good practice to have it there so I left it.

**Pull request closed without merging**